pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -543,6 +543,17 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 	return out, nil
 }
 
+// GetProxyLocality returns the locality where the proxy runs.
+func (c *Controller) GetProxyLocality(proxy *model.Proxy) string {
+	proxyIP := proxy.IPAddresses[0]
+	locality := ""
+	pod := c.pods.getPodByIP(proxyIP)
```

Comment:
I hope this is a map lookup and not a very costly operation

Reply:
Can optimize